### ISSUE TITLE
Multiply operator not merging documents as expected

### ISSUE THREAD

**Issue by inianv** [mikefarah/yq#729]
On `yq version 4.5.1` I'm trying to merge these two YAML files

**master.yaml**

```yaml
containers:
  - name: project-a
    items:
      - CCC
  - name: project-z
    items:
      - CCC
      - DDD
```

and

**update.yaml**

```yaml
- name: project-z
  items:
    - CCC
    - EEE
```

I was hoping the following would work

```bash
yq ea 'select(fi==0) * {"containers":select(fi==1)}' master.yaml update.yaml
```

but it incorrectly produces two records for `project-z` under `containers` . 

```yaml
containers:
  - name: project-a
    items:
      - CCC
  - name: project-z
    items:
      - CCC
      - DDD
  - name: project-z
    items:
      - CCC
      - EEE
```

But if the `update.yaml` contained `project-a` the same command works. For e.g.

```yaml
- name: project-a
  items:
    - CCC
    - EEE
```

it works by appending records as expected.

**Comment by inianv** [comment]
Also @mikefarah Could you also provide a feedback on https://github.com/mikefarah/yq/issues/722#issuecomment-777984510 again? Thanks.

**Comment by mikefarah** [comment]
Fixed in 4.6.0 - merge no longer deeply merges arrays by default.
